File: Search/LinearSearch/linear_search.js
// The function takes an array and the element as input
// Returns the position if the element is found
// Otherwise, returns -1
function linear_search(array, element) {
  const n = array.length;
  for (i = 0; i < n; i++) {
    if (array[i] == element)
      return i;
  }
  return -1;
}


const arr = [4, 12, 7, 90, 88, 14];
const element = 90;
const position = linear_search(arr, element);

if (position == -1)
  console.log("Element not found!");
else
  console.log("Element found at: " + position);

File: Search/LinearSearch/C/linear_search.c
#include <stdio.h>

int search(int arr[], int n, int key)
{
  int i;
  for (i = 0; i < n; i++)
  {
    if (arr[i] == key)
      break;
  }

  return (i == n) ? -1 : i;
}

int main()
{
  int size;
  printf("Enter the size of the array\n");
  scanf("%d\n",&size);
  int arr[size],key;
  int n = sizeof(arr) / sizeof(int);
  printf("Enter the elememts of the array\n");
 for(int i=0;i<n;i++){
    scanf("%d\n",arr[i]);
  }
  printf("Enter the element you want to search");
  scanf("%d\n",&key);
  int res = search(arr, n, key);
  
  if (res == -1)
    printf("%d is not found!\n", key);
  else
    printf("%d is found at position %d.\n", key, res + 1);
  
  return 0;
}
